src/Oscoin/Crypto/Address/Serialisation.hs

```diff
@@ -0,0 +1,237 @@
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -fno-warn-orphans  #-}
+module Oscoin.Crypto.Address.Serialisation
+    ( DeserializeError(..)
+    , serializeAddress
+    , deserializeAddress
+    ) where
+
+import           Oscoin.Prelude
+
+import qualified Oscoin.Crypto.Address.Bech32 as Bech32
+import           Oscoin.Crypto.Address.Internal
+import           Oscoin.Crypto.Hash
+
+import           Codec.CBOR.ByteArray (toSliced)
+import qualified Codec.CBOR.Decoding as CBOR
+import qualified Codec.Serialise as CBOR
+import qualified Codec.Serialise.Encoding as CBOR
+import           Control.Monad.Except (Except, liftEither, throwError)
+import           Control.Monad.Fail (fail)
+import           Data.ByteArray (ByteArrayAccess)
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Builder as BL
+import           Data.Tagged
+
+
+{------------------------------------------------------------------------------
+  Poor man's parsing primitives, to be kept here or moved into a separate
+  package.
+------------------------------------------------------------------------------}
+
+data DeserializeError =
+      NotEnoughInput  ByteString
+      -- ^ The deserialisation failed as the decoder was expecting more
+      -- input.
+    | DeserializePayloadError CBOR.DeserialiseFailure
+      -- ^ The deserialisation process failed while deserialising the
+      -- 'AddressPayload'.
+    | UnknownProtocolVersion Word8
+      -- ^ When decoding the 'ProtocolVersion', we encountered an unknown tag.
+    | UnknownAddressType Word8
+      -- ^ When decoding the 'AddressType'', we encountered an unknown tag.
+    | UnknownAddressFormat Word8
+      -- ^ When decoding the 'AddressFormat, we encountered an unknown tag.
```

Comment:
```suggestion
      -- ^ When decoding the 'AddressFormat', we encountered an unknown tag.
```
Not sure how the missing quote ended up one line above

Reply:
hah, was wondering the same 😂

Reply:
35cf933